pkg/kv/kvserver/replica_consistency.go

```diff
@@ -643,6 +642,60 @@ func (*Replica) sha512(
 		return err
 	}
 
+	rangeKeyVisitor := func(rangeKV storage.MVCCRangeKeyValue) error {
+		// Rate limit the scan through the range.
+		err := limiter.WaitN(ctx,
+			int64(len(rangeKV.RangeKey.StartKey)+len(rangeKV.RangeKey.EndKey)+len(rangeKV.Value)))
+		if err != nil {
+			return err
+		}
+
+		if snapshot != nil {
+			// Add (a copy of) the range key into the debug message.
+			rkv := roachpb.RaftSnapshotData_RangeKeyValue{
+				Timestamp: rangeKV.RangeKey.Timestamp,
+			}
+			alloc, rkv.StartKey = alloc.Copy(rangeKV.RangeKey.StartKey, 0)
+			alloc, rkv.EndKey = alloc.Copy(rangeKV.RangeKey.EndKey, 0)
+			alloc, rkv.Value = alloc.Copy(rangeKV.Value, 0)
+			snapshot.RangeKV = append(snapshot.RangeKV, rkv)
+		}
+
+		// Encode the length of the start key and end key.
+		binary.LittleEndian.PutUint64(intBuf[:], uint64(len(rangeKV.RangeKey.StartKey)))
+		if _, err := hasher.Write(intBuf[:]); err != nil {
+			return err
+		}
+		binary.LittleEndian.PutUint64(intBuf[:], uint64(len(rangeKV.RangeKey.EndKey)))
+		if _, err := hasher.Write(intBuf[:]); err != nil {
+			return err
+		}
+		binary.LittleEndian.PutUint64(intBuf[:], uint64(len(rangeKV.Value)))
+		if _, err := hasher.Write(intBuf[:]); err != nil {
+			return err
+		}
+		if _, err := hasher.Write(rangeKV.RangeKey.StartKey); err != nil {
+			return err
+		}
+		if _, err := hasher.Write(rangeKV.RangeKey.EndKey); err != nil {
+			return err
+		}
+		legacyTimestamp = rangeKV.RangeKey.Timestamp.ToLegacyTimestamp()
```

Comment:
For point keys we do:
		binary.LittleEndian.PutUint64(intBuf[:], uint64(len(unsafeValue)))
		if _, err := hasher.Write(intBuf[:]); err != nil {
			return err
		}
regardless of the key being value or tombstone. We now have a local timestamp in the tombstone what would make it potentially "different" from the other one. Do we need to do the same here as range tombstone is also a value?

Reply:
We already do:

https://github.com/cockroachdb/cockroach/blob/ee57d72f56e9f0c5240618a9b20dde9a71b18731/pkg/kv/kvserver/replica_consistency.go#L673-L676

https://github.com/cockroachdb/cockroach/blob/ee57d72f56e9f0c5240618a9b20dde9a71b18731/pkg/kv/kvserver/replica_consistency.go#L695-L696

Am I missing something?

Reply:
Oh, I see it in file more, but review shows the old versions which I assume is before rebase to the new MVCCRangeKeyValue era.